### PR TITLE
Add needs64bitPositions() to Layer and use it in existing layers

### PR DESCRIPTION
**Pull request by georgios-uber**
For #1937 

#### Background
See RFC.

#### Change List
- Add new method for deciding if we need 64-bit array

### CODE REVIEW

File: modules/core/src/lib/layer.js [hunk 1]
```diff
@@ -193,6 +193,10 @@ export default class Layer extends Component {
     return false;
   }
 
+  needs64bitPositions() {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js]: Why do we need a wrapper for `is64bitEnabled` function? Can we consolidate?

Isn't the key change we need to remove the `if (this.props.fp64)` check from `is64bitEnabled`?

File: docs/whats-new.md [hunk 2]
```diff
@@ -21,6 +21,9 @@ It is now possible to globally set WebGL parameters (controlling how the GPU ren
 
 Pixel sizes in `LineLayer`, `IconLayer` and `TextLayer` now match their HTML/SVG counterparts.
 
+### `is64bitEnabled()` deprecated
```
Comment (ibgreen) [inline docs/whats-new.md]: Let's move this to Upgrade Guide. This is a very small change that only affects 64bit layer writers, if mentioned at all it should come at the end of this v6.0 section.

File: modules/core/src/lib/layer.js [hunk 3]
```diff
@@ -180,6 +180,11 @@ export default class Layer extends Component {
   }
 
   is64bitEnabled() {
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js]: Move to end of class, after `// DEPRECATED` comment.

File: modules/core/src/lib/layer.js [hunk 4]
```diff
@@ -194,6 +199,10 @@ export default class Layer extends Component {
     return false;
   }
 
+  use64bitPositions() {
+    return this.props.fp64;
```
Comment (ibgreen) [inline modules/core/src/lib/layer.js]: Assume you are going to change this in next PR, otherwise should be same as above for now.

Comment (georgios-uber) [reply modules/core/src/lib/layer.js]: Yes. I will change in next PR.

File: modules/core/src/lib/layer.js (hunk 3 above)
Comment (georgios-uber) [reply modules/core/src/lib/layer.js]: OK

File: docs/whats-new.md (hunk 2 above)
Comment (georgios-uber) [reply docs/whats-new.md]: OK

**Pull request merged**